modules/javafx.web/src/test/java/test/javafx/scene/web/SVGTest.java

```diff
@@ -137,4 +137,155 @@
             assertTrue("Color should be white:" + pixelAt100x100, isColorsSimilar(Color.WHITE, pixelAt100x100, 1));
         });
     }
+
+    @Test public void testSVGRectRenderingWithMask() {
+        loadContent(
+                    "<html>\n" +
+                    "  <body style='margin: 0px 0px;'>\n" +
+                    "    <svg width='200' height='200'>\n" +
+                    "      <defs>\n" +
+                    "        <linearGradient id='Gradient'>\n" +
+                    "          <stop offset='0' stop-color='white' stop-opacity='0' />\n" +
+                    "          <stop offset='1' stop-color='white' stop-opacity='1' />\n" +
+                    "        </linearGradient>\n" +
+                    "        <mask id='Mask'>\n" +
+                    "          <rect x='0' y='0' width='200' height='200' fill='url(#Gradient)' />\n" +
+                    "        </mask>\n" +
+                    "      </defs>\n" +
+                    "      <rect x='0' y='0' width='200' height='200' fill='green' />\n" +
+                    "      <rect x='0' y='0' width='200' height='200' fill='red' mask='url(#Mask)' />\n" +
+                    "    </svg>\n" +
+                    "  </body>\n" +
+                    "</html>"
+        );
+        submit(() -> {
+            final WebPage webPage = WebEngineShim.getPage(getEngine());
+            assertNotNull(webPage);
+            final BufferedImage img = WebPageShim.paint(webPage, 0, 0, 200, 200);
```

Comment:
This is added to the (preexisting) problem of calling `paint` on the wrong thread. In this case, it doesn't seem to cause any additional problems, and other tests in this same class do it, so we can fix this in the follow-on issue that is already filed, [JDK-8252596](https://bugs.openjdk.java.net/browse/JDK-8252596).

Reply:
moved tests to system test. also consolidated all tests into one.